beetsplug/absubmit.py

```diff
@@ -0,0 +1,159 @@
+import tempfile
+
+import distutils
+import requests
+
+from beets import plugins
+from beets import util
+from beets import ui
+
+
+class ABSubmitError(Exception):
+    """Base exception for all excpetions this plugin can raise."""
+
+
+class FatalABSubmitError(ABSubmitError):
+    """Raised if the plugin is not able to start."""
+
+
+class AnalysisABSubmitError(ABSubmitError):
+    """Raised if analysis of file fails."""
+
+
+class SubmitABSubmitError(ABSubmitError):
+    """Raised if submitting data fails."""
+
+
+def call(args):
+    """Execute the command and return its output.
+
+    Raise a AnalysisABSubmitError on failure.
+    """
+    try:
+        return util.command_output(args)
+    except subprocess.CalledProcessError as e:
+        raise AnalysisABSubmitError(
+            u"{0} exited with status {1}".format(args[0], e.returncode)
+        )
+
+
+class AcousticBrainzSubmitPlugin(plugins.BeetsPlugin):
+
+    def __init__(self):
+        super(AcousticBrainzSubmitPlugin, self).__init__()
+
+        self.config.add({'extractor': u''})
+
+        self.extractor = self.config['extractor'].as_str()
+        if self.extractor:
+            self.extractor = util.normpath(self.extractor)
+            # Expicit path to extractor
+            if not os.path.isfile(self.extractor):
+                raise FatalABSubmitError(
+                    u'extractor command does not exist: {0}'.
+                    format(self.extractor)
+                )
+        else:
+            # Implicit path to extractor, search for it in path
+            # TODO how to check for on Windows?
+            self.extractor = 'streaming_extractor_music'
+            try:
+                call([self.extractor])
+            except OSError:
+                raise FatalABSubmitError(
+                    u'no extractor command found: install "{0}"'.
+                    format(self.extractor)
+                )
+            except AnalysisABSubmitError:
+                # Extractor found, will exit with an error if not called with
+                # the correct amount of arguments.
+                pass
+            # Get the executable needed to calculate the sha1 hash.
+            self.extractor = distutils.spawn.find_executable(self.extractor)
+
+    supported_formats = {'mp3', 'ogg', 'oga', 'flac', 'mp4', 'm4a', 'm4r',
+                         'm4b', 'm4p', 'aac', 'wma', 'asf', 'mpc', 'wv',
+                         'spx', 'tta', '3g2', 'aif', 'aiff', 'ape'}
+
+    base_url = 'https://acousticbrainz.org/api/v1/{mbid}/low-level'
+
+    def commands(self):
+        cmd = ui.Subcommand(
+            'absubmit',
+            help=u'calculate and submit AcousticBrainz analysis'
+        )
+        cmd.func = self.command
+        return [cmd]
+
+    def command(self, lib, opts, args):
+        # Get items from arguments
+        items = lib.items(ui.decargs(args))
+        # Get no_submit option.
+        # TODO get a should submit option from the command line.
```

Comment:
Yes, this was originally planned, but I didn't find it very usefull, but forgot to remove the comment.